daemons/compute/lib-cpp/client.h

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <string>
+#include <memory>
+#include <vector>
+
+class Status;
+class Workflow;
+
+class CreateRequest;
+class CreateResponse;
+class StatusRequest;
+class StatusResponse;
+class CancelRequest;
+class CancelResponse;
+class DeleteRequest;
+class DeleteResponse;
+class ListRequest;
+class ListResponse;
+
+class DataMoverClient {
+    public:
+        
+        DataMoverClient(const std::string &target);
+        ~DataMoverClient();
+
+        Status create(const Workflow &workflow, const CreateRequest &request, CreateResponse *response);
+        Status status(const Workflow &workflow, const StatusRequest &request, StatusResponse *response);
+        Status cancel(const Workflow &workflow, const CancelRequest &request, CancelResponse *response);
+        Status delete_(const Workflow &workflow, const DeleteRequest &request, DeleteResponse *response);
+        Status list (const Workflow &workflow, const ListRequest &request, ListResponse *response);
+
+    private:
+        void *data_;
+};
+
+class Status {
+    public:
+        Status(bool ok, int error_code, std::string error_message);
+
+        bool ok() const { return ok_; }
+        int error_code() const { return error_code_; }
+        std::string error_message() const { return error_message_; }
+
+    private:
+        bool ok_;
+        int error_code_;
+        std::string error_message_;
+};
+
+class Workflow {
+    public:
+        Workflow(std::string name, std::string namespace_);
+
+        std::string name() const { return name_; }
+        std::string namespace_() const { return namespace__; }
+
+    private:
+        friend DataMoverClient;
+
+        std::string name_;
+        std::string namespace__;
```

Comment:
Identifiers with a double underscore anywhere are reserved.
https://en.cppreference.com/w/cpp/language/identifiers

Reply:
How do you know this stuff?

Reply:
At my prior employer I read nearly every book written by Scott Meyers and Herb Sutter.

Reply:
Nerd